core/src/main/java/org/infinispan/xsite/statetransfer/XSiteStateTransferManager.java

```diff
@@ -97,7 +97,19 @@
     */
    void becomeCoordinator(String siteName);
 
-   XSiteStateProvider getStateProvider();
+   /**
+    * @deprecated since 10.0. Get the {@link XSiteStateProvider} from the {@link org.infinispan.factories.ComponentRegistry}
+    */
+   @Deprecated
+   default XSiteStateProvider getStateProvider() {
+      throw new UnsupportedOperationException();
+   }
 
-   XSiteStateConsumer getStateConsumer();
+   /**
+    * @deprecated since 10.0. Get the {@link XSiteStateConsumer} from the {@link org.infinispan.factories.ComponentRegistry}
+    */
+   @Deprecated
+   default XSiteStateConsumer getStateConsumer() {
+      throw new UnsupportedOperationException();
```

Comment:
No need to deprecate these methods as they were introduced in `10.0.0.Beta4` by https://issues.jboss.org/browse/ISPN-10322. Just remove them :slightly_smiling_face: